api/handlers/datasources.py

```diff
@@ -0,0 +1,51 @@
+import database.database
+from fastapi import APIRouter, Depends, Response
+from pydantic import BaseModel
+from pymongo.database import Database
+
+router = APIRouter()
+
+
+class Datasource(BaseModel):
+    name: str
+    accountName: str
+    containerName: str
+    description: str
+
+
+# class GetDatasourcesResponse(BaseModel):
+#    datasources: list
+
+
+@router.post("/api/datasources", status_code=201)
+def create_datasource(
+    datasource: Datasource,
+    response: Response,
+    db: Database = Depends(database.database.db),
+):
+    """
+    Create a new datasource. The datasource will be henceforth identified by accountName/containerName which
+    must be unique or this function will return a 400.
+    """
+    if db.datasources.find_one(
```

Comment:
Do we have a unique index on account name and container name? Just thinking of a race condition where two requests are made at the same time, this would return null and two would be added

Reply:
We currently do not have given the current level of usage of the app I think this would not be a huge risk thart we are into but definetly something to keep in mind for the future. 
Curreently we do not have any databse model not as part of the application anyway.